src/sentry/services/hybrid_cloud/notifications/impl.py

```diff
@@ -136,12 +134,18 @@ def get_settings_for_user_by_projects(
             )
         ]
 
-    def remove_notification_settings(self, *, actor_id: int, provider: ExternalProviders) -> None:
+    def remove_notification_settings(
+        self, *, team_id: Optional[int], user_id: Optional[int], provider: ExternalProviders
+    ) -> None:
         """
         Delete notification settings based on an actor_id
         There is no foreign key relationship so we have to manually cascade.
         """
-        NotificationSetting.objects._filter(target_ids=[actor_id], provider=provider).delete()
+        team_ids = [team_id] if team_id else None
+        user_ids = [user_id] if user_id else None
+        NotificationSetting.objects._filter(
+            team_ids=team_ids, user_ids=user_ids, provider=provider
+        ).delete()
```

Comment:
What would happen if both `team_id` and `user_id` were non-null arguments? At a glance, it looks like it might filter down to an empty query set and delete nothing.

Do we expect it always to be called with only one non-null ID? If so, I'd rather break it up into two separate service methods, such as
```python
   def remove_notification_settings_for_user(self, *, user_id: int, provider: ExternalProviders) -> None: pass
   def remove_notification_settings_for_team(self, *, team_id: int, provider: ExternalProviders) -> None: pass
```